### ISSUE TITLE
Filter by year

### ISSUE THREAD

**Issue by aniskasmi** [arodygin/datatables-bundle#19]
Hi, do you have tips or an move to pass customer parameter from request to handle to made a filter per Year, on my page i have an select input with 5 last years, when user click an parameter, ?year=2021 pass on link, how to get them to handle for add $query->where("p.year = $year")

**Comment by webinarium** [comment]
Hi,

There was no way to pass something extra, so... I implemented it as a new feature and released it in the new version 2.5.0. :)

With the new version, handlers receive a new optional parameter `array $context`, which can be used to pass anything you need. For example, in your controller it could be something like this:
```
public function action(Request $request, DataTablesInterface $datatables): JsonResponse
{
    $context = [
        'year' => $request->get('year')
    ];

    $results = $datatables->handle($request, 'months', $context);

    return $this->json($results);
}
```

and in the handler:
```
public function handle(DataTableQuery $request, array $context = []): DataTableResults
{
    if ($context['year'] === 2021) ...
}
```

Hope this helps.

**Comment by aniskasmi** [comment]
Thx a lot for example for futur users :) 

Twig
```
var table = $('#data-table-sse').DataTable({
            serverSide: true,
            ajax: '{{ path('datatable_project', {'year': selectYear}) }}' // Pass from controller and $request->get('year')
        });
```
Action
```
public function projectAction(Request $request, DataTablesInterface $datatables): JsonResponse
    {
        $context = [
            'year' => $request->get('year')
        ];
    
        $results = $datatables->handle($request, 'project', $context);
    
        return $this->json($results);
    }
```
Handle
```
// Query to get requested entities.
        $query = $this->repository->createQueryBuilder('p')
            ->where('YEAR(p.addedAt) = :year')
            ->setParameter('year', $context['year'])
        ;
```

After you can custom them with search by isActive to get deleted entry ;)
